src/main/java/gregtech/api/recipes/Recipe.java

```diff
@@ -60,6 +64,25 @@ public static String formatChanceValue(int outputChance) {
 
     private final RecipePropertyStorage recipePropertyStorage;
 
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Recipe recipe = (Recipe) o;
+        return duration == recipe.duration &&
+                EUt == recipe.EUt && inputs.equals(recipe.inputs) &&
```

Comment:
Should a separate check be added for not consumed items, or are those part of `inputs` at this point?